source/extensions/filters/network/redis_proxy/proxy_filter.cc

```diff
@@ -24,9 +24,25 @@ ProxyFilterConfig::ProxyFilterConfig(
       stat_prefix_(fmt::format("redis.{}.", config.stat_prefix())),
       stats_(generateStats(stat_prefix_, scope)),
       downstream_auth_username_(
-          Config::DataSource::read(config.downstream_auth_username(), true, api)),
-      downstream_auth_password_(
-          Config::DataSource::read(config.downstream_auth_password(), true, api)) {}
+          Config::DataSource::read(config.downstream_auth_username(), true, api)) {
+
+  auto downstream_auth_password =
+      Config::DataSource::read(config.downstream_auth_password(), true, api);
+  if (!downstream_auth_password.empty()) {
+    downstream_auth_passwords_.emplace_back(downstream_auth_password);
+  }
+
+  if (config.extra_downstream_auth_passwords_size() > 0) {
+    downstream_auth_passwords_.reserve(downstream_auth_passwords_.size() +
+                                       config.extra_downstream_auth_passwords().size());
+    for (const auto& source : config.extra_downstream_auth_passwords()) {
+      const auto p = Config::DataSource::read(source, true, api);
+      if (!p.empty()) {
```

Comment:
Why does this version have the empty check? Seems like it should be the same as above?

Reply:
We were having password empty check (which was on `std::string`) and fail AUTH. To maintain the same behavior, we need to have empty `std:vector` if the provided password is empty. We could probably ask the `Config::Datasource:read` to not allow empty string if we want to avoid empty check.